openwisp_controller/config/api/serializers.py

```diff
@@ -149,6 +151,16 @@ def _create_config(self, device, config_data):
             raise serializers.ValidationError({'config': error.messages})
 
     def _update_config(self, device, config_data):
+        if (
+            config_data.get('backend') == app_settings.BACKENDS[0][0]
```

Comment:
this should compare to the default value of the `Config.backend` field, which if I remember correctly can be modified via `OPENWISP_CONTROLLER_DEFAULT_BACKEND`.